docs/developer/ceed.md

```diff
@@ -0,0 +1,206 @@
+
+# RDycore SWE physics with CEED
+
+Here we describe the RDycore's implemetation of 2D shallow water equation (SWE) that uses [libCEED](https://github.com/CEED/libCEED). The combination of PETSc and libCEED provides RDDycore algorithimic and hardware protability.
+
+- RDycore uses PETSc's `TS` solvers that provide support for multiple time-integrators such as forward euler, RK4, etc., which can be selected at run time (e.g. `--ts_type euler`, `--ts_type rk4`, etc). 
+- The use libCEED allows RDycore to compute the RHS-function for the explicit time-integrators on CPU or GPU, which can also be selected at runtime via
+     - For CPUs: `-ceed /cpu/self`
+     - For NVIDIA GPUs: `-ceed /gpu/cuda -dm_vec_type cuda`
+     - For AMD GPUs: `-ceed /gpu/hip -dm_vec_type hip`
+
+## Example Mesh
+
+Let's consider an example 3x2 mesh that consists of
+
+- 6 cells: `c00` to `c05`
+- 17 edges: `e00` to `e16`
+    - Internal edges: `e04`, `e05`, `e07`, `e08`, `e09`, `e11`, and `e12`.
+    - Boundary edges: `e00`, `e01`, `e02`, `e03`, `e06`, `e10`, `e13`, `e14`, `e15`, and `e16`.
+- 12 vertices: `v00` to `v11`
+
+```text
+v08---e14---v09---e15---v10---e16---v11
+ |           |           |           |
+ |           |           |           |
+e10   c03   e11   c04   e12   c05   e13
+ |           |           |           |
+ |           |           |           |
+v04---e07---v05---e08---v06---e09---v07
+ |           |           |           |
+ |           |           |           |
+e03   c00   e04   c01   e05   c02   e06
+ |           |           |           |
+ |           |           |           |
+v00---e00---v01---e01---v02---e02---v03
+
+```
+
+For the mesh shown above, the prognostic variables of the 2D SWE are saved in a strided PETSc Vec (`X`).
+The block size of `X` is `3` corresponding to the following prognostic variables:
+
+- Height (`h`),
+- Momentum in x-dir (`hu`), and
+- Momentum in y-dir (`hv`).
+The size of `X` will `6 * 3` where `6` corresponds to number of cells in the mesh. The layout of `X` will be as follows:
+
+```text
+X = [[h0 hu0 hv0] [h1 hu1 hv1] ... [h5 hu5 hv5]]
+```
+
+## SWE physics with libCEED
+
+The libCEED version of RDycore's explicit time-integrator of the SWE solver has two `CeedOperator`:
```

Comment:
`CeedOperator`s:

Reply:
Let's also link to [CeedOperator](https://libceed.org/en/latest/api/CeedOperator/) here.